File: scripts/repo-paths.mjs
#!/usr/bin/env node

import fs from 'node:fs';
import path from 'node:path';

export const AEONITE_CTS_ROOT_ENV = 'AEONITE_CTS_ROOT';
export const AEON_TOOLING_ROOT_ENV = 'AEON_TOOLING_ROOT';
export const AEON_TOOLING_PRIVATE_ROOT_ENV = 'AEON_TOOLING_PRIVATE_ROOT';
export const AEONITE_SPECS_ROOT_ENV = 'AEONITE_SPECS_ROOT';

export function getRepoRoot() {
  return path.resolve(path.dirname(new URL(import.meta.url).pathname), '..');
}

export function getFamilyRoot() {
  return path.resolve(getRepoRoot(), '..', '..');
}

export function getAeoniteCtsRoot() {
  return process.env[AEONITE_CTS_ROOT_ENV]
    || path.resolve(getFamilyRoot(), 'aeonite-org', 'aeonite-cts', 'cts');
}

export function getAeonToolingRoot() {
  const publicDefault = path.resolve(getFamilyRoot(), 'altopelago', 'aeon-tooling');
  return process.env[AEON_TOOLING_ROOT_ENV]
    || process.env[AEON_TOOLING_PRIVATE_ROOT_ENV]
    || (fs.existsSync(publicDefault)
      ? publicDefault
      : path.resolve(getFamilyRoot(), 'altopelago', 'aeon-tooling-private'));
}

export function getAeonToolingPrivateRoot() {
  return getAeonToolingRoot();
}

export function getAeoniteSpecsRoot() {
  return process.env[AEONITE_SPECS_ROOT_ENV]
    || path.resolve(getFamilyRoot(), 'aeonite-org', 'aeonite-specs');
}

export function withRepoPathEnv(baseEnv = process.env) {
  return {
    ...baseEnv,
    [AEONITE_CTS_ROOT_ENV]: getAeoniteCtsRoot(),
    [AEON_TOOLING_ROOT_ENV]: getAeonToolingRoot(),
    [AEON_TOOLING_PRIVATE_ROOT_ENV]: getAeonToolingRoot(),
    [AEONITE_SPECS_ROOT_ENV]: getAeoniteSpecsRoot(),
  };
}

export function resolveCTSPath(candidate, cwd = process.cwd()) {
  if (!candidate) return getAeoniteCtsRoot();

  const resolved = path.resolve(cwd, candidate);
  const normalized = candidate.replaceAll('\\', '/');
  const parts = normalized.split('/');
  const ctsIndex = parts.lastIndexOf('cts');
  if (ctsIndex === -1) return fs.existsSync(resolved) ? candidate : candidate;

  const remainder = parts.slice(ctsIndex + 1).join('/');
  return path.resolve(getAeoniteCtsRoot(), remainder);
}

File: scripts/repo_paths.py
from __future__ import annotations

from pathlib import Path
import os

AEONITE_CTS_ROOT_ENV = "AEONITE_CTS_ROOT"
AEON_TOOLING_PRIVATE_ROOT_ENV = "AEON_TOOLING_PRIVATE_ROOT"
AEONITE_SPECS_ROOT_ENV = "AEONITE_SPECS_ROOT"


def get_repo_root() -> Path:
    return Path(__file__).resolve().parents[1]


def get_family_root() -> Path:
    return get_repo_root().parents[1]


def get_aeonite_cts_root() -> Path:
    return Path(
        os.environ.get(
            AEONITE_CTS_ROOT_ENV,
            str(get_family_root() / "aeonite-org" / "aeonite-cts" / "cts"),
        )
    )


def get_aeon_tooling_private_root() -> Path:
    return Path(
        os.environ.get(
            AEON_TOOLING_PRIVATE_ROOT_ENV,
            str(get_family_root() / "altopelago" / "aeon-tooling-private"),
        )
    )


def get_aeonite_specs_root() -> Path:
    return Path(
        os.environ.get(
            AEONITE_SPECS_ROOT_ENV,
            str(get_family_root() / "aeonite-org" / "aeonite-specs"),
        )
    )


def repo_path_env(base: dict[str, str] | None = None) -> dict[str, str]:
    env = dict(base or os.environ)
    env.setdefault(AEONITE_CTS_ROOT_ENV, str(get_aeonite_cts_root()))
    env.setdefault(AEON_TOOLING_PRIVATE_ROOT_ENV, str(get_aeon_tooling_private_root()))
    env.setdefault(AEONITE_SPECS_ROOT_ENV, str(get_aeonite_specs_root()))
    return env
